lib/puppet/feature/base.rb

```diff
@@ -77,6 +77,7 @@ def self.is_implemented
           true
         rescue LoadError => err
           Puppet.debug("CreateSymbolicLink is not available")
+          Puppet.deprecation_warning("Windows 2003 and below versions are deprecated, and will be officially unsupported in the next major Puppet release.  Please plan accordingly and see http://links.puppetlabs.com/deprecate-windows-2003 for more information.")
```

Comment:
* I was working from the incorrect assumption that features were lazy-evaluated.  That made me think I was preventing the addition of extra code to call the feature resolution code, but it turns out that's not how it works.  So we can segregate out to a `windows_2003` or similar feature and introduce some redundant code if Puppet is the place to introduce the warning. (I also wanted to make sure I hit a code path that was used by `agent`, `apply`, `config`, etc)
* Text was mostly a placeholder at the moment until we could polish out the messaging.  I should have mentioned that with the PR.  Pointeds noted @kylog - thanks.

So that said, I'm not sure this is the route we want to go given deprecation message noise.  Might be better to skin this cat inside https://github.com/puppetlabs/puppet_for_the_win instead.